src/directives/leaflet.js

```diff
@@ -24,9 +23,9 @@ angular.module("leaflet-directive", []).directive('leaflet',
         transclude: true,
         template: '<div class="angular-leaflet-map"><div ng-transclude></div></div>',
         controller: function ($scope) {
-            _leafletMap = $q.defer();
+            this._leafletMap = $q.defer();
```

Comment:
That is not exactly true the reference is needed so the directives are invoked in the correct order and at the correct time. The directive can not proceed without a reference to a map. But yes the map reference should be unique to the link function.